components/experimental/src/units/ratio.rs

```diff
@@ -0,0 +1,193 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use core::{
+    ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Sub, SubAssign},
+    str::FromStr,
+};
+
+use num_bigint::BigInt;
+use num_rational::Ratio;
+use num_traits::Signed;
+use num_traits::{One, Pow, Zero};
+
+use super::provider::{Base, SiPrefix};
+
+// TODO: implement AsULE for IcuRatio and use it in Data module.
+// TODO: add test cases for IcuRatio.
+/// A ratio type that uses `BigInt` as the underlying type.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct IcuRatio(Ratio<BigInt>);
```

Comment:
Does it need to be `pub`? Can you make it `pub(crate)`?

Reply:
We need to have it pub because we will use it in the datagen and also as an output of the converter.

Reply:
I added a TODO comment because I do not want to move the test cases for now. we can do it in another PR.